app/js/controllers/main.js

```diff
@@ -104,22 +82,70 @@ export default class MainController extends Controller {
       this.appendChildController.mainController = this;
       this.copyMoveController.mainController = this;
 
-      console.log('[Customizer] Lazy-initialized modules');
+      console.log('[Customizer] Initialized controllers and views');
       resolve();
     });
+  }
+
+  _cleanUp() {
+    window.removeEventListener('visibilitychange', this._visibilitychangeHandler);
+
+    this.editController.mainController = null;
```

Comment:
This is getting very very goopy, and it would be better to store these in an array or something and iterate through them. We should really file a follow-up and deal with that.